File: legal-shield/src/types/resources.ts
import type { LegalCategory } from "@/types/analysis";

export interface LegalResource {
  id: string;
  name: string;
  type: "clinic" | "lawyer-referral" | "law-firm";
  state: string;
  city?: string;
  county?: string;
  phone?: string;
  website?: string;
  address?: string;
  coverage?: string;
  categories: LegalCategory[];
  source: string;
}

File: backend/app/schemas/resources.py
from __future__ import annotations

from pydantic import BaseModel


class LegalResourceOut(BaseModel):
    id: str
    name: str
    type: str
    state: str
    city: str | None = None
    county: str | None = None
    phone: str | None = None
    website: str | None = None
    address: str | None = None
    coverage: str | None = None
    categories: list[str]
    source: str = "curated"

    model_config = {"from_attributes": True}
